modules/tinymce/src/core/main/ts/fmt/RemoveFormat.ts

```diff
@@ -306,6 +306,13 @@ const removeFormatInternal = (ed: Editor, format: RemoveFormatPartial, vars?: Fo
           }
         }
 
+        // keep style="list-style-type: none" on <li>s
+        if (name === 'style' && NodeType.matchNodeNames([ 'li' ])(elm) && dom.getStyle(elm, 'list-style-type') === 'none') {
```

Comment:
I realised this is going to cause the logic on line 322 to not run, which might leave a case where the `data-mce-style` attribute would remain. As such we probably need to remove both the `style` and `data-mce-style` attribute in this short-circuit case or alternatively move this short circuit case to below line 322

Note: I suspect the `dom.setStyle` call will clean this up anyways, but I'm not confident on relying on that and as such I think we should explicitly clean it up.

- [ ] Ensure the `data-mce-style` attribute is also cleaned up properly.